src/TwigComponent/src/Command/TwigComponentDebugCommand.php

```diff
@@ -0,0 +1,335 @@
+    }
+
+    protected function configure(): void
+    {
+        $this
+            ->setDefinition([
+                new InputArgument('name', InputArgument::OPTIONAL, 'A component name or part of the component name'),
+                new InputOption('dir', null, InputOption::VALUE_REQUIRED, 'Show all components with template in a specific directory'),
+            ])
+            ->setHelp(<<<'EOF'
+The <info>%command.name%</info> display all the Twig components in your application.
+
+To list all components:
+
+    <info>php %command.full_name%</info>
+
+To get specific information about a component, specify its name:
+
+    <info>php %command.full_name% Alert</info>
+
+To find all component templates within a directory, specifying the <info>--dir</info> option:
+
+    <info>php %command.full_name% --dir=bar/foo</info>
+EOF
+            );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $io = new SymfonyStyle($input, $output);
+        $name = $input->getArgument('name');
+        $directory = $input->getOption('dir');
+
+        if (\is_string($name)) {
+            $component = $this->findComponentName($io, $name, $input->isInteractive());
+            if (null === $component) {
+                $io->error(sprintf('Unknown component "%s".', $name));
+
+                return Command::FAILURE;
+            }
+
+            $this->displayComponentDetails($io, $component);
+
+            return Command::SUCCESS;
+        }
+
+        $components = $this->findComponents($directory);
+        $this->displayComponentsTable($io, $components);
+
+        return Command::SUCCESS;
+    }
+
+    public function complete(CompletionInput $input, CompletionSuggestions $suggestions): void
+    {
+        if ($input->mustSuggestArgumentValuesFor('name')) {
+            $suggestions->suggestValues(array_keys($this->findComponents()));
+        }
+    }
+
+    private function findComponentName(SymfonyStyle $io, string $name, bool $interactive): ?string
+    {
+        $components = [];
+        foreach ($this->componentClassMap as $componentName) {
+            if ($name === $componentName) {
+                return $name;
+            }
+            if (str_contains($componentName, $name)) {
+                $components[$componentName] = $componentName;
+            }
+        }
+
+        foreach ($this->findAnonymousComponents() as $componentName) {
+            if (isset($components[$componentName])) {
+                continue;
+            }
+            if ($name === $componentName) {
+                return $name;
+            }
+            if (str_contains($componentName, $name)) {
+                $components[] = $componentName;
+            }
+        }
+
+        if ($interactive && \count($components)) {
+            return $io->choice('Select one of the following component to display its information', array_values($components), 0);
+        }
+
+        return null;
+    }
```

Comment:
nice method

Reply:
Looking at it now, could be simplified (it did more at a moment, using metadata)